### DISCUSSION TITLE
Welcome to amplipi Discussions!

### DISCUSSION THREAD

**Discussion by linknum23** [micro-nova/AmpliPi#48] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by coatmaker618** [comment]
Hello!  I'm a nerd who came across this project in about 87 different ways.  As somebody who has fallen far down the rabbit hole of selfhosting things, I too have been looking for whole home audio that's not closed off.  The fact that this is open source, and not $10k is AWESOME :)

I did have a few questions, I hope this is the right place to ask them!

1. What was thought process behind the design decision to provide only analog inputs?  Why not have some option for digital audio input as well (TOSLink, HDMI, ect)?

2. You mention using this for home theatre use.  Generally those are 5.1 or 7.1 (or more) channels, but this box seems to have 4 pairs of stereo audio as inputs.

3. Could these 4 inputs be used in combination as a single home theatre system?  That is to say, what latency/slew rate are you seeing between channels?  And about about syncing between boxes? Will 2 boxes be completely synced or, if they're nearby, would one notice the difference?

4. Judging by your kickstarter page (https://www.kickstarter.com/projects/micro-nova/amplipi-home-audio-system/description)  you are using 2 different DACs, #1 (TI's PCM5102A) for input #1 & #2 (Cmedia's CM6206) for input #2-4   what was the design choice here?  I'm not familiar with the DAC market myself, but naively why would you not use a single 4 source stereo DAC OR 4x1 source stereo DAC?

That said, all of these questions are about a small component of this whole project & creating a platform is immensely important.  And please don't take this as criticism but rather my own ignorance having NOT done this! 

Thank you :)

**Reply by mabowers** [reply]
Hey @coatmaker618,

Thank you for your interest and your kind words about AmpliPi! I'm on the AmpliPi team at MicroNova and I would be more than happy to answer your questions:

1. The design choice to go with analog RCA inputs was driven by a few different factors: <br>
- Firstly, we were uncertain of what mix of inputs to provide. There are a lot of possible ports (optical and coaxial S/PDIF, HDMI, XLR, BNC, RCA, etc.) and combinations thereof. For the first version of AmpliPi we decided to keep things simple with RCA. The nice thing about RCA ports is that they are pretty universal and small adapters for all manner of digital standards to RCA are readily available for $20-40. If we get a lot of feedback like "oh I really wish it had ports for standard X" then we can definitely take that into account for future revisions.<br>
- Some of the digital standards are a little tricky... For example, S/PDIF _can_ carry uncompressed PCM which is no problem to work with, but it _can also_ carry closed-source codecs that require extra licensing terms and royalties (e.g. DTS, Dolby) that make them difficult to integrate with an open-source product. We could simply choose to support PCM only, but it may be confusing for users who see a S/PDIF port and expect it to just work with the surround sound from their Blu-ray player. HDMI has similar challenges with the HDCP DRM. (Google "HDCP 2.2" to see some of the truly ridiculous headaches the latest HDMI revisions are creating for soundbars and AV receivers.) These aren't insurmountable obstacles; it is _possible_ that we can still create an open source product that supports these standards, by compartmentalizing them into optional, closed-source software and hardware components that live inside a larger open source product. But it is going to require more investigation.

2. While AmpliPi was primarily designed to control multiple stereo zones throughout a home, there are home theater uses we also envisioned. The CM6206 USB DAC can be used to output 5.1 (L front, R front, center, L rear, R rear, woofer - a total of 6 channels) using three of the stereo outputs. We have not heavily tested this configuration, but it is possible. AmpliPi itself can also be used as an HTPC. It has an HDMI output port (with H.264 decoding) and USB for adding a remote control (e.g. a FLIRC receiver). You could also use AmpliPi in conjunction with an existing home theater system, using it to pipe the stereo output throughout the house to enjoy media as you move about the kitchen, garage, laundry room, etc. The beautiful thing about AmpliPi being open source is that users have the _freedom to tinker with everything_ to meet their exact needs.

3. The input channels (whether they come from the four analog RCA inputs, or from the Pi's DACs) each have a _direct analog path_ to the amplifier outputs. The inputs pass through analog multiplexers that introduce negligible latency. Similarly, the signal path to each amplifier output _on the expansion boxes_ is a buffered analog signal with negligible latency. There will no perceptible delay between different amplifier outputs, whether they are on the same box or daisy-chained boxes. There will also be no perceptible delay between two or more _different_ analog RCA inputs. (There _may_ be a delay between the I2S and USB DACs, but we have not characterized this, nor would this be a typical use case.) Have a look at [this diagram](https://github.com/micro-nova/AmpliPi/blob/master/hw/diagrams/signal-diagram.png) or [the schematics](https://github.com/micro-nova/AmpliPi/tree/master/hw/sch) to see what this signal path looks like.

4. The Raspberry Pi has only a single I2S output that we wired to the TI PCM5102A DAC to provide the first stereo output channel. (We've read that it _might_ be possible to time-multiplex several I2S DACs with the Raspberry Pi. It is not clear if we could run all four DAC outputs this way, and there may be other tradeoffs associated with this configuration. This would require further investigation. To keep things clean and simple, we opted to go with a single I2S DAC for the first revision of AmpliPi.) The remaining three stereo output channels are provided over USB with the CM6206 DAC; the neat thing about this is that it is a multi-channel DAC that, with some tweaks to the AmpliPi software, can enable the use of 5.1 surround sound as discussed in item 3 above. We felt that these two DACs provided a flexible mix of audio outputs from the Pi. One other thing to note is that the TI PCM5102A specifications are 32-bit @ 384kHz, while the CM6206 specifications are 16-bit @ 48kHz. Our team have not been able to perceive any difference between the two DACs, but for folks with the right setup (high bitrate, high sample rate, lossless audio files, high-end speakers and superhuman hearing) they will appreciate the first channel from the TI DAC (and this is why we didn't use the CM6206 for all four channels.)

Thanks again for your interest in AmpliPi. If you have any other questions, comments, or suggestions, don't hesitate to mention them :)

**Reply by coatmaker618** [reply]
Thanks for the very detailed reply @mabowers! 

I certainly have some homework now :p

And those points make sense--I was not considering the risk of consumer confusion for only certain protocols working.  As you said, "this only works sometimes" is problematic for the user if you can't easily explain it to them.

One thing that's still not clear to me is how synchronized having two of these on the same network will be.  My assumption is "very" but I just wanted to verify that from your perspective/tests.  Or to see if you looked at that use case.

**Reply by coatmaker618** [reply]
One other question I wanted to ask-- in my mind, the software should be able to sync 2 channel audio between two Rasp-PIs on the same network VERY WELL (see previous question).  If this is not the case, then this question is moot.

Otherwise, judging by the descriptions I've read it should be fairly easy to build a single channel version as a demo.  Digikey has the TI Amp for about $5 for 1 unit, you connect that to the I2S & bam, you get audio (unamplified).  Are there any reference documents for making such a device?

**Reply by mabowers** [reply]
AmpliPi "synchronizes" the audio between two boxes via the expansion connection on the back of the box (chained from the AmpliPi main unit to one or more AmpliPi expansion units.) It does not do this over the network. This should be possible, but it is not the use case that we designed AmpliPi for. I'm sure there is existing open source software designed for network audio sync that can be integrated with AmpliPi. If this is a feature you'd like to see (and maybe help contribute to!) open [an issue](https://github.com/micro-nova/AmpliPi/issues) :)

For a demo/test version of AmpliPi that you can run right away, have a look at the instructions in [this document](https://github.com/micro-nova/AmpliPi/blob/master/DEVELOPING.md). You can grab a CMedia-based multi-channel USB DAC (similar to the DAC we use for channels 2-4) for about $25, hook it up to a Raspberry Pi (doesn't need to be a CM3+), and get up and running with the AmpliPi software!

**Reply by mabowers** [reply]
@coatmaker618, we've managed to add a S/PDIF input and additional aux input on the final version of the hardware:
https://www.kickstarter.com/projects/micro-nova/amplipi-home-audio-system/posts/3127934

Here is an updated signal diagram. See additions in blue:

![Signal Diagram 2](https://user-images.githubusercontent.com/22546058/111033678-69732980-83e0-11eb-9221-7a4b63e10392.png)
